ray-operator/test/e2eautoscaler/raycluster_autoscaler_test.go

```diff
@@ -509,3 +509,63 @@ func TestRayClusterAutoscalerGPUNodesForCPUTasks(t *testing.T) {
 		})
 	}
 }
+
+// This test verifies that the autoscaler can launch nodes to fulfill ray.autoscaler.sdk.request_resources from the user program.
+func TestRayClusterAutoscalerSDKRequestResources(t *testing.T) {
+	tc := tests[0] // Use the default autoscaler config
+
+	test := With(t)
+	g := gomega.NewWithT(t)
+
+	// Create a namespace
+	namespace := test.NewTestNamespace()
+
+	// Mount the call_request_resources.py script as a ConfigMap
+	scriptsAC := newConfigMap(namespace.Name, files(test, "call_request_resources.py"))
+	scripts, err := test.Client().Core().CoreV1().ConfigMaps(namespace.Name).Apply(test.Ctx(), scriptsAC, TestApplyOptions)
+	g.Expect(err).NotTo(gomega.HaveOccurred())
+	LogWithTimestamp(test.T(), "Created ConfigMap %s/%s successfully", scripts.Namespace, scripts.Name)
+
+	test.T().Run("Test ray.autoscaler.sdk.request_resources", func(_ *testing.T) {
+		groupName := "request-group"
+
+		rayClusterSpecAC := rayv1ac.RayClusterSpec().
+			WithEnableInTreeAutoscaling(true).
+			WithRayVersion(GetRayVersion()).
+			WithHeadGroupSpec(rayv1ac.HeadGroupSpec().
+				WithRayStartParams(map[string]string{"num-cpus": "0"}).
+				WithTemplate(tc.HeadPodTemplateGetter())).
+			WithWorkerGroupSpecs(rayv1ac.WorkerGroupSpec().
+				WithReplicas(0).
+				WithMinReplicas(0).
+				WithMaxReplicas(3).
```

Comment:
```suggestion
				WithMaxReplicas(5).
```
Make the max a little higher and we should test we only get 3 nodes created.

Reply:
Addressed. Thanks!